### PR TITLE
General: Fix finding of last version

### PR DESCRIPTION
**Pull request by antirotor**
## Bug

There was a bug that affected finding of last OP version. This resulted in invalid behaviour mainly in headless mode, where it tries to reinstall already installed version.

### CODE REVIEW

**Comment by BigRoy** [comment]
This would require deploying new executables/installers, yes? :)

Were you able to reproduce the original issue of it trying to do a force install in headless mode? And did this avoid it on your end? Just wondering how "safe" it is for me to just throw this PR into production to see if it'd would fix the issue or whether this might do more harm.

**Comment by antirotor** [comment]
This still needs little refactor to move the compatibility check out...

**Comment by antirotor** [comment]
> This would require deploying new executables/installers, yes? :)

yes, this will need new build


> Were you able to reproduce the original issue of it trying to do a force install in headless mode? And did this avoid it on your end? Just wondering how "safe" it is for me to just throw this PR into production to see if it'd would fix the issue or whether this might do more harm.

I haven't reproduced it, but find similar issue with headless mode that lead me to this. The re-install logic is sound I think, problem is that it can't find the latest version correctly so it is trying to reinstall it.

But please wait with deployment until the compatibility check is refactored.

**Comment by BigRoy** [comment]
> But please wait with deployment until the compatibility check is refactored.

If you could ping me when that's the case, please do so! :)

File: igniter/bootstrap_repos.py [hunk 1]
```diff
@@ -515,17 +503,27 @@ def get_versions_from_directory(
...
+                          item.name.split(".")[
+                              1] == installed_version.minor  # noqa: E051
+                if item.is_dir() and ver_dir:
+                    _versions = OpenPypeVersion.get_versions_from_directory(
+                        item)
+                    if _versions:
+                        openpype_versions.append(_versions)
+            except IndexError:
+                pass
+            # if file exists, strip extension, in case of dir don't.
```
Comment (BigRoy) [inline igniter/bootstrap_repos.py:526]: Wouldn't the `item.name` match exactly with `f"{installed_version.major}.{installed_version.minor}"`?
So wouldn't this just be `if item.name == f"{installed_version.major}.{installed_version.minor}"` or alike?

```
major_minor = f"{installed_version.major}.{installed_version.minor}"
if item.is_dir() and item.name == major_minor:
    _versions = OpenPypeVersion.get_versions_from_directory(item)
    openpype_versions.extend(_versions)
```

And shouldn't we also "continue" if we've already identified it as a major minor subfolder, so add
```
    continue
```

File: igniter/bootstrap_repos.py [hunk 2]
```diff
@@ -629,11 +615,7 @@ def get_latest_version(
 
         all_versions.sort()
         latest_version: OpenPypeVersion
```
Comment (iLLiCiTiT) [inline igniter/bootstrap_repos.py:617]:
```suggestion
```

**Comment by BigRoy** [comment]
We have deployed this in production now. (Since this requires the build/install to be updated too I've actually batch removed the old install and then batch deployed the new to make sure it's using this newer build logic.)

So far so good in Deadline. If anything pops up with errors I'll report back.

**Comment by antirotor** [comment]
> Added UI message if installed version is not compatible with zip and is not launched in headless mode.

It works.

**Pull request merged**